### ISSUE TITLE
Sidekiq issues after postgres- and server updates

### ISSUE THREAD

**Issue by tofuwabohu** [mastodon/mastodon#26783]
### Steps to reproduce the problem

1. Upgrade Postgres from 13-15
2. Upgrade debian from bullseye to bookworm
3. Wait poeple to post
...

### Expected behaviour

The posts should show up on people's home feeds

### Actual behaviour

The home feed is empty, but the posts are visible on the author's profiles

### Detailed description

Hi,
we upgraded to a new postgres version (13->15), since there were new posts in federated timeline we thought everything was fine. 
Then we upgraded debian from bullseye to bookworm.

It turns out: home feeds are not updated and notifications are missing, and the search throws 502 bad gateway. While searching the user does not work, in the web client domain.tld/@user@other.domain leads to the correct profile.
Moreover, domain.tld/sidekiq gives me the error page.
and sidekiq keeps restarting

However, the length of the ingress queue is kept < 10. so this seems fine.

But we have:
Sep 03 22:42:43 Mastodon bundle[620291]: /home/mastodon/live/vendor/bundle/ruby/3.0.0/gems/llhttp-ffi-0.4.0/lib/llhttp/parser.rb:79: [BUG] Segmentation fault at 0x0000000000000000

Also, for me, /admin/dashboard results in "We're sorry, but something went wrong on our end.", but /sidekiq works. For the other admin, it's vice versa.

Any idea how to further investigate?

### Mastodon instance

subversive.zone

### Mastodon version

v4.1.4

### Technical details

If this is happening on your own Mastodon server, please fill out those:
- Ruby version: ruby 3.0.4p208
- Node.js version: v18.13.0

**Comment by ClearlyClaire** [comment]
This does not look like it's related to the PostgreSQL update, but to the system library update and possibly Ruby build issues. This might be Ruby mixing malloc and jemalloc, or something similar.

Are you using Debian's Ruby, or `rbenv`? Did you rebuild Ruby after the Debian update? Did you rebuild any of the gems after the Debian update?

**Comment by tofuwabohu** [comment]
Thank you. We solved the issue, I'll document  the process in case someone has the same issue. We're using Debian's Ruby.

We tried `bundle pristine` and `bundle install` first, which ran through but didn't have an effect. We then tried `RAILS_ENV=production bundle exec gem pristine --all`, which errored with
```
Building native extensions. This could take a while...
ERROR:  While executing gem ... (Gem::Ext::BuildError)
    ERROR: Failed to build gem native extension.

    current directory: /home/mastodon/live/vendor/bundle/ruby/3.0.0/gems/racc-1.6.2/ext/racc/cparse
/home/mastodon/.rbenv/versions/3.0.6/bin/ruby -I /home/mastodon/.rbenv/versions/3.0.6/lib/ruby/3.0.0 -r ./siteconf20230904-886714-6xh184.rb extconf.rb
Ignoring racc-1.6.2 because its extensions are not built. Try: gem pristine racc --version 1.6.2
```
Running `RAILS_ENV=production bundle install --deploymen` apparently the extension ( `Installing racc 1.6.2 with native extensions`), but running the bundle exec from above again resulted in the same error.

We then reinstalled ruby, de- and reinstalled bundle, `bundle install`, `yarn install`, `RAILS_ENV=production bundle exec rails yarn:install assets:precompile`. This resulted in the Ingress queue getting super large, but after a while it was processed and everything seems fine now.
